docs/sdk-configuration.md

```diff
@@ -0,0 +1,492 @@
+  public void doLogic() {
+    Span span = tracer.spanBuilder("logic").startSpan();
+    try (Scope ignored = span.makeCurrent()) {
+      Thread.sleep(1000);
+    } finally {
+      span.end();
+    }
+  }
+}
+```
+
+## The global instance of the SDK
+
+A built instance is convenient to use in most Java apps because of dependency injection. Because it
+has a easy-to-reason lifecycle, being tied into the dependency ordering (even if dependency 
+injection happened to be done manually through constructor invocation), we encourage application
+developers to only use it.
+
+However, there are corner cases where an instance cannot be injected. The famous example is MySQL -
+MySQL interceptors are initialized by calling a default constructor, and there is no way to pass a
+built instance of a signal provider. For this case, we must store an SDK instance into a global
+variable. To reduce gotchas for application developers, the first SDK that is instantiated is stored
+automatically in the global variable.
+
+Before an SDK has been set, access to the global SDK will throw an exception and will not return a
+default. The global can only be used after an SDK has been configured. SDKs must be configured early
+in an application to ensure it applies to all of the logic in the app, and this will generally be
+ensured by the configuration framework such as Spring. For application developers, this restriction
+should not have any effect one way or the other in the vast majority of cases.
+
+MySQL is the only known corner case that requires the global SDK instance. If such a corner case
+didn't exist, we may not even support it in the first place.
+
+See special note with Java Agent below though.
+
+## Telemetry within SDK components
+
+SDK components, such as exporters or remote samplers, may want to emit telemetry for their own
+processing. However, the SDK components must be initialized before the SDK can be fully built. We do
+not support partially built SDK because one cannot reason about the behavior of it. Similarly we
+do not support using the global instance of the SDK before it has been built. Therefore, SDK
+components that require `OpenTelemetry` must accept it lazily. This is a restriction, but given
+such components are rarely developed by application developers, and generally developed by either
+framework authors or OpenTelemetry maintainers, this restriction is deemed reasonable.
+
+If this mechanism was built into the SDK, it may look like
+
+```java
+interface OpenTelemetryComponent {
+  default void setOpenTelemetry(OpenTelemetry openTelemetry) {}
+}
+interface SpanExporter extends OpenTelemetryComponent {
+}
+public class BatchExporter implements SpanExporter {
+  
+  private volatile Tracer tracer;
+  
+  @Override
+  public void setOpenTelemetry(OpenTelemetry openTelemetry) {
+    traer = openTelemetry.getTracerProvider().get("spanexporter");
+  }
+  
+  @Override
+  public void export() {
+    tracer.spanBuilder("export").startSpan();
```

Comment:
obviously would need a null check here for the tracer, but that's a nit, obviously